docs/sequence/PISPLinkingOTPApi.puml

```diff
@@ -0,0 +1,431 @@
+POST /linking/request-consent/6789/authenticate
+{
+  authToken: '123456'
+}
+end note
+
+activate PISP
+activate PISP_TP_OUT
+
+PISP_TP_OUT -> PISP_TP_OUT: const model = await loadFromKVS({key: 6789})
+rnote right of PISP_TP_OUT: state: webAuthenticationChannelResponseRecieved or OTPAuthenticationChannelResponseRecieved
+PISP_TP_OUT -> PISP_LM: model.authenticate()
+
+activate PISP_LM
+
+PISP_LM -> PISP_LM: ThirdpartyRequests.patchConsentRequests()
+PISP_LM -> Switch: PATCH /consentRequests/6789
+rnote right of PISP_LM #LightBlue
+PATCH /consentRequests/6789
+FSIOP-Source: pispa
+FSIOP-Destination: dfspa
+{
+  authToken: '124356'
+}
+end note
+
+activate Switch
+Switch --> PISP_LM: 202 Accepted
+deactivate PISP_LM
+Switch -> DFSP_TP_IN: PATCH /consentRequests/6789
+activate DFSP_TP_IN
+
+DFSP_TP_IN -> DFSP_TP_IN: const model = await loadFromKVS({key: 6789})
+activate DFSP_LM
+rnote right of DFSP_TP_IN: state: consentRequestValidatedAndStored
+DFSP_TP_IN -> DFSP_LM: model.validateAuthToken()
+DFSP_TP_IN --> Switch: 202 Accepted
+deactivate Switch
+deactivate DFSP_TP_IN
+
+DFSP_LM -> DFSP_LM: DFSPBackendRequests.validateAuthToken()
+rnote right of DFSP_LM #LightBlue
+Do we need two backend endpoints for validating
+web authTokens and OTP authTokens? Or is a DFSP expected to
+validate both cases with one endpoint?
+
+POST /validateAuthToken
+{
+  consentRequestId: '6789'
+  authChannel: model.data.authChannel,
+  authToken: '124356'
+}
+end note
+rnote right of DFSP_LM: state: authTokenValidated
+
+DFSP_LM -> DFSP_LM: const consentId = uuid()
+DFSP_LM -> DFSP_LM: model.grantConsent()
+DFSP_LM -> DFSP_LM: ThirdpartyRequests.postConsents()
+rnote right of DFSP_LM: state: consentGranted
+rnote right of DFSP_LM #LightBlue
+It's important to save the model with the consentId from this point onwards!
+Requests will not contain a consentRequestId in the upcoming requests.
+This may require changing `PersistentModel` to have a secondary key it can
+store the model with.
+end note
+DFSP_LM -> DFSP_LM: this.saveToKVS({key: '1a2b3c4d'})
+
+activate Switch
+DFSP_LM -> Switch: POST /consents
+
+rnote left of DFSP_LM #LightBlue
+POST /consents
+FSIOP-Source: dfspa
+FSIOP-Destination: pispa
+{
+  consentId: 1a2b3c4d
+  consentRequestId: 6789
+  scopes: model.data.scopes
+}
+end note
+Switch --> DFSP_LM: 202 ACCEPTED
+deactivate DFSP_LM
+activate PISP_TP_IN
+Switch ->  PISP_TP_IN: POST /consents
+PISP_TP_IN -> PISP_TP_IN: const model = await loadFromKVS({key: 6789})
+activate PISP_LM
+PISP_TP_IN -> PISP_LM: model.registerCredential()
+PISP_TP_IN --> Switch: 202 Accepted
+deactivate PISP_TP_IN
+deactivate Switch
+rnote right of PISP_LM: state: consentRecieved
+rnote right of PISP_LM #LightBlue
+It's important to save the model with the consentId from this point onwards!
+Requests will not contain a consentRequestId in the upcoming requests.
+This may require changing `PersistentModel` to have a secondary key it can
+store the model with.
```

Comment:
I see two possible solutions:
- we can create the second instance of the model using existing state data but with a new key and call saveToKVS
- or we can refactor save/load methods and add an optional parameters to specify special/new key.